tests/admin.test.ts

```diff
@@ -104,16 +105,67 @@ describe('retroactive attendance submission', () => {
       asStaff: true,
     };
 
-    await adminController.submitAttendanceForUsers(request, proxyUser);
-
-    const userResponse = await userController.getUser({ uuid: user.uuid }, proxyUser);
-    const staffUserResponse = await userController.getUser({ uuid: staffUser.uuid }, proxyUser);
-    const activityResponse = await userController.getCurrentUserActivityStream(user);
-    const staffActivityResponse = await userController.getCurrentUserActivityStream(staffUser);
+    await adminController.submitAttendanceForUsers(request, admin);
 
+    const userResponse = await userController.getUser({ uuid: user.uuid }, admin);
     expect(userResponse.user.points).toEqual(event.pointValue);
+
+    const staffUserResponse = await userController.getUser({ uuid: staffUser.uuid }, admin);
     expect(staffUserResponse.user.points).toEqual(event.pointValue + event.staffPointBonus);
+
+    const activityResponse = await userController.getCurrentUserActivityStream(user);
     expect(activityResponse.activity[1].type).toEqual(ActivityType.ATTEND_EVENT);
+
+    const staffActivityResponse = await userController.getCurrentUserActivityStream(staffUser);
     expect(staffActivityResponse.activity[1].type).toEqual(ActivityType.ATTEND_EVENT_AS_STAFF);
   });
 });
+
+describe('email retrieval', () => {
+  test('gets all the emails of stored users', async () => {
+    const conn = await DatabaseConnection.get();
+    const users = UserFactory.create(5);
+    const emails = users.map((user) => user.email.toLowerCase());
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+
+    await new PortalState()
+      .createUsers([...users, admin])
+      .write();
+
+    const response = await ControllerFactory.admin(conn).getAllEmails(admin);
+    expect(expect.arrayContaining(response.emails)).toEqual([...emails, admin.email]);
+  });
+});
+
+describe('bonus points submission', () => {
+  test('updates points and activity to the users in the bonus request', async () => {
+    const conn = await DatabaseConnection.get();
+    const [userNotGettingBonus, ...users] = UserFactory.create(5);
+    const emails = users.map((user) => user.email.toLowerCase());
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+
+    await new PortalState()
+      .createUsers([...users, userNotGettingBonus, admin])
+      .write();
+
+    const bonus = {
+      description: 'Test addition of bonus points',
+      users: emails,
+      points: 200,
+    };
+
+    const userController = ControllerFactory.user(conn);
+
+    const createBonusResponse = await ControllerFactory.admin(conn).addBonus({ bonus }, admin);
+    expect(createBonusResponse.emails).toEqual(expect.arrayContaining(emails));
+
+    for (let u = 0; u < users.length; u += 1) {
+      const user = users[u];
+      const getUserResponse = await userController.getUser({ uuid: user.uuid }, admin);
+      expect(getUserResponse.user.points).toEqual(200);
```

Comment:
Let's also check that the users' activities are correct.

Reply:
Resolved in 6688e90.